### PR TITLE
fix(SR-5): protect `external-api` from SSRF

### PR DESCRIPTION
**Pull request by ChristianMcc**
## Introduction:
External API was vulnerable to certain SSRF attacks

## Resolution:
Add validations to any user input variables used in further http(s) requests

NOTE: this PR has flagged repeated code warning by sonarcloud, I'm not very worried about this but on Monday I'll look at using test cases to reduce a bit of the repetition on these tests!

### CODE REVIEW

File: external-api/api-tests/v1/ordnance-survey.api-test.ts [hunk 1]
```diff
@@ -92,5 +92,33 @@ describe('/ordnance-survey', () => {
       expect(status).toEqual(200);
       expect(body.results).toBeDefined();
     });
+
+    it('returns a 400 if you provide an postcode', async () => {
```
Comment (abhi-markan) [inline external-api/api-tests/v1/ordnance-survey.api-test.ts:96]: `provide an invalid postcode` please.

File: external-api/api-tests/v1/party-db.api-test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { app } from '../../src/createApp';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
```
Comment (abhi-markan) [inline external-api/api-tests/v1/party-db.api-test.ts:2]: Can we remove inline rules please?

File: external-api/api-tests/v1/party-urn.api-test.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { app } from '../../src/createApp';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
```
Comment (abhi-markan) [inline external-api/api-tests/v1/party-urn.api-test.ts:2]: Remove inline please.

File: external-api/src/utils/inputValidations.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { regexes } from '../constants/regex.constants';
+import validator from 'validator';
+
+export const isValidInput = (regex: RegExp, input: string): boolean => regex.test(input);
```
Comment (abhi-markan) [inline external-api/src/utils/inputValidations.ts:4]: Add function documentations please.

File: external-api/src/v1/controllers/premium-schedule.controller.ts [hunk 5]
```diff
@@ -122,6 +123,11 @@ export const getPremiumSchedule = async (req: Request, res: Response) => {
     };
   }
 
+  if (!validUkefId(premiumScheduleParameters.facilityURN.toString())) {
```
Comment (abhi-markan) [inline external-api/src/v1/controllers/premium-schedule.controller.ts:126]: `facilityURN` do not have prefixed `00` compare to standard UKEF ID (String) `00300..`.
Will the validator pass 8 digit IDs without suffixed `00`?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, only few minor comments to address please.

File: external-api/src/v1/controllers/premium-schedule.controller.ts (hunk 5 above)
Comment (ChristianMcc) [reply external-api/src/v1/controllers/premium-schedule.controller.ts:126]: Yep it takes 8-10 long so it accepts both (just like the APIM MDM endpoint does)

File: external-api/api-tests/v1/ordnance-survey.api-test.ts (hunk 1 above)
Comment (abhi-markan) [reply external-api/api-tests/v1/ordnance-survey.api-test.ts]: @ChristianMcc  missed. Thanks.

File: external-api/api-tests/v1/party-db.api-test.ts (hunk 2 above)
Comment (abhi-markan) [reply external-api/api-tests/v1/party-db.api-test.ts]: @ChristianMcc  missed. Thanks.

File: external-api/api-tests/v1/party-urn.api-test.ts (hunk 3 above)
Comment (abhi-markan) [reply external-api/api-tests/v1/party-urn.api-test.ts]: @ChristianMcc  missed. Thanks.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Looks good and nice test refactorisation very clean.
Only three comments to address which were missed. Thanks

**Review by abhi-markan — DISMISSED** [review]

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**